src/common/core/cachedmetacore.js

```diff
@@ -0,0 +1,363 @@
+                var root = core.getRoot(node),
+                    oldpath = core.getPath(node),
+                    moved = oldcore.moveNode(node, parent);
+
+                if (root.metaElements[oldpath]) {
+                    delete root.metaElements[oldpath];
+                    root.metaElements[core.getPath(moved)] = moved;
+                }
+
+                return moved;
+            };
+
+            //additional inquiry functions
+            core.isMetaNode = function (node) {
+                var root = core.getRoot(node);
+                if (root.metaElements && root.metaElements[core.getPath(node)]) {
+                    return true;
+                }
+
+                return false;
+            };
+
+            core.getAllMetaNodes = function (node) {
+                var root = core.getRoot(node);
+
+                if (root.metaElements) {
+                    return root.metaElements;
+                }
+
+                return [];
+            };
+
+            core.isAbstract = function (node) {
+                return core.getRegistry(node, 'isAbstract') === true;
+            };
+
+            core.isConnection = function (node) {
+                var validPtrNames = oldcore.getValidPointerNames(node);
+
+                return validPtrNames.indexOf('dst') !== -1 && validPtrNames.indexOf('src') !== -1;
+            };
+
+            function sensitiveFilter(validNodes) {
+                var i;
+
+                i = validNodes.length;
+                while (i--) {
+                    if (core.isConnection(validNodes[i]) || core.isAbstract(validNodes[i])) {
+                        validNodes.splice(i, 1);
+                    }
+                }
+            }
+
+            //parameters
+            // node - the node in question
+            // children - the current children of the node, so that multiplicity can be checked
+            // sensitive - if true the function do not return the connection and abstract types
+            // multiplicity - if true the function filters out possibilities that fail multiplicity check
+            // aspect - if given the function also filters out valid children type meta nodes based on aspect rule
+            core.getValidChildrenMetaNodes = function (parameters) {
+                var validNodes = [],
+                    node = parameters.node,
+                    metaNodes = core.getRoot(node).metaElements,
+                    keys = Object.keys(metaNodes || {}),
+                    i, j,
+                    typeCounters = {},
+                    children = parameters.children || [],
+                    rules,
+                    inAspect,
+                    temp;
+
+                rules = oldcore.getChildrenMeta(node) || {};
+
+                for (i = 0; i < keys.length; i += 1) {
+                    temp = metaNodes[keys[i]];
+                    while (temp) {
+                        if (rules[oldcore.getPath(temp)]) {
```

Comment:
nope, I just thought it is safer as it may be overridden in this layer, and then it can give back something else (though it is not too likely)